azurelinuxagent/common/protocol/wire.py

```diff
@@ -935,74 +935,56 @@ def check_wire_protocol_version(self):
-                    # wrong.  This is why we try HostPlugin then direct.
-                    try:
-                        host = self.get_host_plugin()
-                        host.put_vm_status(self.status_blob,
-                                           ext_conf.status_upload_blob,
-                                           ext_conf.status_upload_blob_type)
-                        return
-                    except HttpError:
-                        pass
-                    except Exception as e:
-                        if not isinstance(e, ResourceGoneError):
-                            self.report_status_event("Exception uploading status blob (HostPlugin): {0}", ustr(e))
-
-                    try:
-                        if self.status_blob.upload(blob_uri):
-                            from azurelinuxagent.common.event import add_event, WALAEventOperation
-                            add_event(
-                                name=AGENT_NAME,
-                                version=CURRENT_VERSION,
-                                op=WALAEventOperation.ReportStatus,
-                                is_success=True,
-                                message="direct",
-                                log_event=False)
-                            return
-                    except HttpError:
-                        pass
+            blob_uri = ext_conf.status_upload_blob
+            blob_type = ext_conf.status_upload_blob_type
 
-                    return
+            if blob_uri is None:
+                raise ProtocolError("No blob uri found")
+
+            if blob_type not in ["BlockBlob", "PageBlob"]:
+                blob_type = "BlockBlob"
+                logger.verbose("Status Blob type is unspecified, assuming BlockBlob")
+
+            try:
+                self.status_blob.prepare(blob_type)
+            except Exception as e:
+                raise ProtocolError("Exception creating status blob: {0}", ustr(e))
+
+            # Swap the order of use for the HostPlugin vs. the "direct" route.
+            # Prefer the use of HostPlugin. If HostPlugin fails fall back to the
+            # direct route.
+            #
+            # The code previously preferred the "direct" route always, and only fell back
+            # to the HostPlugin *if* there was an error.  We would like to move to
+            # the HostPlugin for all traffic, but this is a big change.  We would like
+            # to see how this behaves at scale, and have a fallback should things go
+            # wrong.  This is why we try HostPlugin then direct.
+            try:
+                host = self.get_host_plugin()
+                host.put_vm_status(self.status_blob,
+                                   ext_conf.status_upload_blob,
+                                   ext_conf.status_upload_blob_type)
+                return
             except Exception as e:
                 if isinstance(e, ResourceGoneError):
```

Comment:
If you're going to treat one exception type differently, then catch it via a separate `except` statement.

Reply:
Should have cleaned this up, thanks.